code/mission/missionparse.cpp

```diff
@@ -2465,6 +2465,14 @@ int parse_create_object_sub(p_object *p_objp)
 		Script_system.RemHookVars({"Ship", "Parent"});
 	}
 
+	if (Game_mode & GM_IN_MISSION && Ship_info[shipp->ship_info_index].is_big_or_huge()) {
+		float mission_time = f2fl(Missiontime);
+		int minutes = (int)(mission_time / 60);
+		int seconds = (int)mission_time % 60;
+
+		mprintf(("%s arrived at %02d:%02d\n", shipp->ship_name, minutes, seconds));
+	}
+
```

Comment:
This correctly belongs inside the previous `if` block, just don't split the condition while you're doing it.

Reply:
I'm not sure I understand, the wingnum and dock wing stuff isn't relevant for the logging, so it should not be under that condition. If I only put it under the in-mission condition, then it would necessarily involve splitting the in-mission condition from the wingum and dock wing conditions.

Reply:
The wingnum and dock wing stuff is very relevant.  Consider that the script hook must fire once and only once for every ship that arrives.  There is one hook here that handles single ships, and one hook in `mission_set_wing_arrival_location` that handles ships in wings.  So if you want your logging statement to fire once and only once for every ship, you should use exactly the same logic.

(Since you don't care about ships in wings, technically the best thing to do is put your block inside the other `if` block, below the script condition, and add `!brought_in_docked_wing` to your block's condition.)

In your very first draft, the logging statement would have fired for every non-wing ship, but it also would have fired for *some* wing ships - not all of them - and it would not have been obvious why some ships were being logged but others weren't.

In the current draft, the most appropriate condition is
```
if (Game_mode & GM_IN_MISSION && Ship_info[shipp->ship_info_index].is_big_or_huge() && (shipp->wingnum == -1)) {
```
but since it's so similar, you might as well enclose it in the previous block with the one modification.

Reply:
Ok, I think I see, I wasn't aware of this wing stuff. So within the same if block and only conditional on big|huge right?

Reply:
Yes, so just:
```
if (Ship_info[shipp->ship_info_index].is_big_or_huge() && !brought_in_docked_wing)
```